rng_guesser/number_guessing_game/src/main.rs

```diff
@@ -0,0 +1,59 @@
+use rand::thread_rng;
+use rand::Rng;
+
+use std::io;
+
+fn main() {
+    println!("Program Started...");
+
+    let start_txt = "
+██████╗     ███╗   ██╗     ██████╗              ██████╗ ██╗   ██╗███████╗███████╗███████╗███████╗██████╗ 
+██╔══██╗    ████╗  ██║    ██╔════╝             ██╔════╝ ██║   ██║██╔════╝██╔════╝██╔════╝██╔════╝██╔══██╗
+██████╔╝    ██╔██╗ ██║    ██║  ███╗            ██║  ███╗██║   ██║█████╗  ███████╗███████╗█████╗  ██████╔╝
+██╔══██╗    ██║╚██╗██║    ██║   ██║            ██║   ██║██║   ██║██╔══╝  ╚════██║╚════██║██╔══╝  ██╔══██╗
+██║  ██║    ██║ ╚████║    ╚██████╔╝            ╚██████╔╝╚██████╔╝███████╗███████║███████║███████╗██║  ██║
+╚═╝  ╚═╝    ╚═╝  ╚═══╝     ╚═════╝              ╚═════╝  ╚═════╝ ╚══════╝╚══════╝╚══════╝╚══════╝╚═╝  ╚═╝                                                                                                                                                                                                                                         
+    ";
+
+    println!("{start_txt}");
+    println!("Made by lucasodev - UNLICENSE");
+    println!(" ");
+    println!(" ");
+    
+    guessing();
+}
+
+fn guessing() {
+    let random_number = get_random_number();
+    
+    let mut guess_right: bool = false;
```

Comment:
Consider the fact that `guess_right` is actually just the result of the comparison `guess == random_number`.

So, if the code was re-written to something like this:
```rust
loop { // loop {} in rust is just a shorthand for `while true {....}`
    let guess = get_user_input();
    if guess == target {
        println!("Yeah, it was right, the number was {target}!");
        return; // By returning here, we immediately get out of the loop and `guessing()`, so no need for an `else` block.
    } 
    println!("Sorry, Your answer was wrong!");
}
```

Reply:
Ohh, yeah, I didn't know that. I have just programmed in rust for two days... will fix it later.